libraries/amqp/include/eosio/amqp/amqp_handler.hpp

```diff
@@ -98,7 +98,10 @@ class amqp {
          std::promise<void> stop_promise;
          auto future = stop_promise.get_future();
          boost::asio::post( *handler_->amqp_strand(), [&]() {
-            stop_promise.set_value();
+            auto& cb = channel_->close();
```

Comment:
If initialization timed out or otherwise failed, it's possible for `channel_` to be null.

Reply:
Add check for null